src/node_platform.h

```diff
@@ -101,6 +101,7 @@ class PerIsolatePlatformData :
   std::shared_ptr<PerIsolatePlatformData> self_reference_;
   uint32_t uv_handle_count_ = 1;  // 1 = flush_tasks_
 
+  v8::Isolate* const isolate_;
```

Comment:
your PR description `Isolate* pointer can generally be looked up from the per-Isolate platform data structure` hints that the pointer is already present there but that is not the case here?

Is there any guidance / best practice around which class could afford to `contain` (make it a field) the isolate pointer as opposed to obtain it from the environment on demand?

Reply:
> your PR description `Isolate* pointer can generally be looked up from the per-Isolate platform data structure` hints that the pointer is already present there but that is not the case here?

@gireeshpunathil Right, what the commit message implies here is that adding the field for lookup is straightforward because it’s a constructor argument.

> Is there any guidance / best practice around which class could afford to `contain` (make it a field) the isolate pointer as opposed to obtain it from the environment on demand?

Most Node.js internals store a pointer to the `Environment` instance, which can be used for efficiently accessing the `Isolate` as well, so usually that’s not a concern. I guess that means that the answer to your question is “no”.

That being said, if “obtain it from the environment on demand” refers specifically to `Isolate::GetCurrent()`, then yes, the V8 team discourages using that function and directly passing along the `Isolate*` pointer, like the commit message says.